Source/Engine/Threading/JobSystem.cpp

```diff
@@ -221,9 +239,18 @@ int64 JobSystem::Dispatch(const Function<void(int32)>& job, int32 jobCount)
 #if JOB_SYSTEM_USE_STATS
     const auto start = Platform::GetTimeCycles();
 #endif
+    const auto label = Platform::InterlockedAdd(&JobLabel, (int64)jobCount) + jobCount;
 
     JobData data;
     data.Job = job;
+    data.JobKey = label;
+
+    auto context = New<JobContext>();
```

Comment:
Could we skip memory dynamic allocations or just reuse them from jobs contexts pool (eg. 8?)

Reply:
Actually I have to revert this change, the dictionary holding the context might get reallocated when it grows so it would invalidate the existing references when waiting for the job to finish.